File: app/src/main/java/com/dougkeen/bart/networktasks/NetworkUtils.java
package com.dougkeen.bart.networktasks;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;

public class NetworkUtils {

    public static OkHttpClient makeHttpClient() {
        return new OkHttpClient.Builder()
                .connectTimeout(CONNECTION_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS)
                .readTimeout(READ_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS)
                .writeTimeout(WRITE_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS)
                .callTimeout(CALL_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS)
                .build();
    }

    static final int CONNECTION_TIMEOUT_MILLIS = 10000;
    static final int READ_TIMEOUT_MILLIS = 15000;
    static final int WRITE_TIMEOUT_MILLIS = 15000;
    static final int CALL_TIMEOUT_MILLIS = 20000;
}

File: app/src/main/kotlin/in/izyum/bart/networktasks/NetworkUtils.kt
package `in`.izyum.bart.networktasks

import okhttp3.OkHttpClient
import java.util.concurrent.TimeUnit

object NetworkUtils {
    @JvmStatic
    fun makeHttpClient(): OkHttpClient = OkHttpClient.Builder()
        .connectTimeout(CONNECTION_TIMEOUT_MILLIS.toLong(), TimeUnit.MILLISECONDS)
        .readTimeout(READ_TIMEOUT_MILLIS.toLong(), TimeUnit.MILLISECONDS)
        .writeTimeout(WRITE_TIMEOUT_MILLIS.toLong(), TimeUnit.MILLISECONDS)
        .callTimeout(CALL_TIMEOUT_MILLIS.toLong(), TimeUnit.MILLISECONDS)
        .build()

    @JvmField
    val CONNECTION_TIMEOUT_MILLIS = 10000

    @JvmField
    val READ_TIMEOUT_MILLIS = 15000

    @JvmField
    val WRITE_TIMEOUT_MILLIS = 15000

    @JvmField
    val CALL_TIMEOUT_MILLIS = 20000
}
